src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/ASEReadCounter.java

```diff
@@ -0,0 +1,286 @@
+        final String contig = alignmentContext.getContig();
+        final long position = alignmentContext.getPosition();
+
+        final char refAllele = (char) referenceContext.getBase();
+
+        final List<VariantContext> VCs = featureContext.getValues(variants);
+        if (VCs != null && VCs.size() > 1)
+            throw new UserException("More then one variant context at position: " + contig + ":" + position);
+        if (VCs == null || VCs.isEmpty())
+            return;
+
+        final VariantContext vc = VCs.get(0);
+        if (!vc.isBiallelic()) {
+            logger.warn("Ignoring site: cannot run ASE on non-biallelic sites: " + vc.toString());
+            return;
+        }
+
+        if (vc.getHetCount() < 1) {
+            logger.warn("Ignoring site: variant is not het at postion: " + contig + ":" + position);
+            return;
+        }
+
+        if (vc.getNAlleles() == 1 || vc.getAlternateAllele(0).getBases().length == 0)
+            throw new UserException("The file of variant sites must contain heterozygous sites and cannot be a GVCF file containing <NON_REF> alleles.");
+
+        final char altAllele = (char) vc.getAlternateAllele(0).getBases()[0];
+
+        final String siteID = vc.getID();
+        final ReadPileup pileup = filterPileup(alignmentContext.getBasePileup(), countType);
+
+        // count up the depths of all and QC+ bases
+        calculateLineForSite(pileup, contig, position, siteID, refAllele, altAllele);
+    }
+
+    @Override
+    public void closeTool() {
+        if (outputStream != null)
+            outputStream.close();
+    }
+
+    protected ReadPileup filterPileup(final ReadPileup originalPileup, final CountPileupType countType) {
+
+        ReadPileup pileupWithDeletions;
+        if (countType.equals(CountPileupType.COUNT_FRAGMENTS_REQUIRE_SAME_BASE))
+            pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(true, true);
+        else if (countType.equals(CountPileupType.COUNT_READS))
+            pileupWithDeletions = originalPileup;
+        else if (countType.equals(CountPileupType.COUNT_FRAGMENTS))
+            pileupWithDeletions = originalPileup.getOverlappingFragmentFilteredPileup(false, true);
+        else
+            throw new UserException("Must use valid CountPileupType");
+
+        return pileupWithDeletions.makeFilteredPileup(p -> !p.isDeletion());
+
+    }
+
+    protected void calculateLineForSite(final ReadPileup pileup, final String contig, final long position, final String siteID, final char refAllele, final char altAllele) {
+
+        int rawDepth = 0, lowBaseQDepth = 0, lowMAPQDepth = 0, refCount = 0, altCount = 0, totalNonFilteredCount = 0, otherBasesCount = 0, improperPairsCount = 0;
+
+        for (final PileupElement base : pileup) {
+            rawDepth++;
+
+            if (base.getRead().isPaired() && (base.getRead().mateIsUnmapped() || !base.getRead().isProperlyPaired())) {
+                improperPairsCount++;
```

Comment:
This line never gets executed, maybe you can add an improper pair to one of your test cases?